codespan-reporting/src/term/views.rs

```diff
@@ -303,21 +335,31 @@ where
                 // Check to see if we need to render any intermediate stuff
                 // before rendering the next line.
                 if let Some((next_line_index, _)) = lines.peek() {
-                    match next_line_index.checked_sub(line_index) {
+                    match next_line_index.checked_sub(*line_index) {
                         // Consecutive lines
                         Some(1) => {}
                         // One line between the current line and the next line
                         Some(2) => {
                             // Write a source line
                             let file_id = labeled_file.file_id;
+                            // get filtered out labels, if there are any
+                            let labels = if let Some((_, line)) = labeled_file
+                                .lines
+                                .iter()
+                                .find(|(index, _)| **index == line_index + 1)
+                            {
+                                &line.multi_labels
+                            } else {
+                                &current_labels
+                            };
```

Comment:
Not really understanding what this line is for. What happens if we just use `&current_labels` in the call to `render_snippet_source` instead of doing this?

Reply:
This is inside a match case that is invoked if the next line that must be rendered is 2 lines away. In this case it does not make sense to hide the next line (making a one line omission) and it is rendered although it was previously filtered out. Therefore the labels on that line have to be retrieved again from the unfiltered `labeled_file.lines`.
Although I must admit, it would probably be cleaner with a `map_or`, I'll fix that.